sycl/source/detail/memory_manager.cpp

```diff
@@ -160,20 +160,41 @@ void *MemoryManager::allocateMemImage(
                              Format);
 }
 
-void *MemoryManager::createSubBuffer(RT::PiMem ParentMem, size_t ElemSize,
-                                     id<3> Offset, range<3> Range,
-                                     std::vector<RT::PiEvent> DepEvents,
-                                     RT::PiEvent &OutEvent) {
+void *MemoryManager::allocateMemSubBuffer(ContextImplPtr TargetContext,
+                                          void *ParentMemObj, size_t ElemSize,
+                                          size_t Offset, range<3> Range,
+                                          std::vector<RT::PiEvent> DepEvents,
+                                          RT::PiEvent &OutEvent) {
   waitForEvents(DepEvents);
   OutEvent = nullptr;
 
+  if (TargetContext->is_host())
+    return static_cast<void *>(static_cast<char *>(ParentMemObj) + Offset);
+
+  size_t SizeInBytes = ElemSize;
+  for (size_t I = 0; I < 3; ++I)
+    SizeInBytes *= Range[I];
+
+  return allocateSubBufferObject(TargetContext, ParentMemObj, Offset,
+                                 SizeInBytes);
+}
+
+void *MemoryManager::allocateSubBufferObject(ContextImplPtr TargetContext,
```

Comment:
Probably it would be better to merge this function with "allocateMemSubBuffer".
It seems TargetContext is not used by the function.

Reply:
Will do